sys/Makefile.dep

```diff
@@ -173,6 +173,7 @@ endif
 
 ifneq (,$(filter netdev_ieee802154_submac,$(USEMODULE)))
   USEMODULE += netdev_ieee802154
+  USEMODULE += netdev_legacy_api
```

Comment:
Huh? Isn't the submac API the new API?

Reply:
No, this is the new API this PR is about: #14660

Reply:
When the porting of the new API is not even done yet and there is already a new new API :sweat_smile:

Reply:
No driver can be the new API, as neither GNRC nor lwIP are compatible with new non-blocking network devices. (They would free the buffer after calling `send()` - which in case of zero-copy drivers such as `stm32_eth` would free a buffer while the DMA is still sending it out. They would also try to call `send()` again without waiting for `TX_COMPLETE`.)

Reply:
The radio HAL and the netdev API are orthogonal to each other.

However, this `request_op()` --> `confirm_op()` pattern is actually taken from the radio HAL (which in term borrows it from the Linux kernel, if I recall correctly) and used for `send()` --> `confirm_send()`. So providing the `netdev` API on top of the IEEE820154 radio HAL will be a bit easier once the network stacks support the new netdev API.

Reply:
Thanks for clarifying!